### PR TITLE
fix build error, update documentation, and add shell lint action

### PR DESCRIPTION
**Pull request by andrewmcodes**
# Bug Fix

## Description

The action-ci test does not like the `action.yml` file currently.

Error:
```
/usr/local/lib/ruby/2.6.0/json/common.rb:156:in `parse': 767: unexpected token at 'File path 'app/**/*.html.haml' does not exist (JSON::ParserError)
110
'
```

## Why should this be added

This will fix the issue causing the syntax verification to fail and also adds some updates to the documentation. Also adds shell lint to hopefully avoid `sh` issues in the future.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] Actions are passing

### CODE REVIEW

File: lib/index.rb [hunk 1]
```diff
@@ -19,7 +19,9 @@ def read_json(path)
   repo: ENV["GITHUB_REPOSITORY_NAME"] || @event_json.dig("repository", "name"),
 }
 
-@haml = "haml-lint " + ENV["INPUT_FILE_PATHS"]
+@input_file_paths = ENV["INPUT_FILE_PATHS"].empty? ? "app/views/" : ENV["INPUT_FILE_PATHS"]
```
Comment (andrewmcodes) [inline lib/index.rb:22]: I guess we technically don't need to add this but I would rather default to `app/views` rather than `app/views/**/*.haml.html`

Comment (chiefpansancolt) [reply lib/index.rb:22]: Fair assumption! better to be safe then sorry

File: action.yml [hunk 2]
```diff
@@ -35,6 +32,10 @@ runs:
   args:
     - ${{ inputs.version }}
     - ${{ inputs.additional_gems }}
+    - ${{ inputs.file_paths }}
```
Comment (chiefpansancolt) [inline action.yml]: Why are we adding these as args? they automatically become ENV variables when run, they just need to be args if referenced in the docker or entrypoint.sh files. Unless I am missing something

File: README.md [hunk 3]
```diff
@@ -71,15 +71,17 @@ jobs:
...
-| additional_gems | false    |                      | Additional Gems can be installed via one line with spaces and commands are supported like a version |
-| config_path     | false    | repo ./              | If the config path is another spot in the repo or not named .haml-lint.yml                          |
-| exclude_paths   | false    |                      | Define a list of paths to exclude from being linted.                                                |
-| fail_level      | false    | 'error'              | Can define 'error' or 'warning' to cause haml-lint to error out on                                  |
+Refer to the [official `haml-lint` documentation](https://github.com/sds/haml-lint#usage) for more information on the `haml-lint` options.
+
+| ****Input Parm Name**** | ****Required**** | ****Default Value**** | ****`haml-lint` option**** |                                                                                  ****Description****                                                                                  |              **Example**              |
```
Comment (chiefpansancolt) [inline README.md:76]: Whats the indication of haml-lint column, little confused by the usage/reference

**Review by chiefpansancolt — COMMENTED** [review]
See comments inline

File: README.md (hunk 3 above)
Comment (andrewmcodes) [reply README.md:76]: Was trying to indicate if it was a haml-lint config option. I can remove it unless you can think of a better way to word this. Basically my intent was to say 
> this is a haml-lint config option, refer to their documentation for more info or for how to use this.

File: action.yml (hunk 2 above)
Comment (andrewmcodes) [reply action.yml]: Good point, I need to read the documentation again. I thought you should specify all of your supported inputs here but I could be wrong.

File: README.md (hunk 3 above)
Comment (chiefpansancolt) [reply README.md:76]: i think then instead of true false use yes/no

File: action.yml (hunk 2 above)
Comment (chiefpansancolt) [reply action.yml]: from what i can tell without them then they still show available when running, but from other useages out there they were only added to args when used in the docker and entrypoint files all others were just inputs, but def double check

**Review by chiefpansancolt — APPROVED** [review]
LGTM

**Pull request merged**